permission_prompt/explariner_navigation.md

```diff
@@ -0,0 +1,89 @@
+
+## Deprecated solutions
+### Service worker
+
+In the [previous discussion](https://github.com/WICG/private-network-access/issues/83), we planned to introduce the permission for
+workers, so that it can be used for non-fetch requests. However, the mixed 
+content check actually happens before service workers, which means it won't work.
+
+## A solution
+
+In this case, we would like to introduce a new [Content Security Policy](https://www.w3.org/TR/CSP3/) to let 
+all documents declare their IP address space.
+
+### New [directive](https://www.w3.org/TR/CSP3/#framework-directives)
+
+Two new Content Security Policy directives are introduced for Private Network Access, 
+`private-address-space` and `local-address-space`.
+
+> Notes: `localhost`, `::1/128` and `127.0.0.1` hosts are regard as secure in mixed 
+> content check, so that we don't need to relax. Benchmarking address 198.18.0.0/15 is regarded as a local IP address in Private Network Access specification but not potentially trustworthy origin in secure contexts spec.
+
+### Policy delivery
+
+[The same as all the other Content Security Policies](https://www.w3.org/TR/CSP3/#policy-delivery), private/local IP address 
+space can be declared as either a HTTP header or a HTML `<meta>` element.
+
+#### HTTP header
+
+To declare private IP addresses:
+
+```text
+Content-Security-Policy: private-address-space private_hosts
+```
+
+Example:
+```text
+Content-Security-Policy: private-address-space example.com *.example.com
+```
+
+To declare local IP addresses:
+
+```text
+Content-Security-Policy: local-address-space local_hosts
+```
+
+Example:
+```text
+Content-Security-Policy: local-address-space localhost
+```
+
+#### <meta> element
+
+```text
+<meta http-equiv="Content-Security-Policy" content="private-address-space example.com local-address-space localhost">
+```
+
+### Reporting
+
+The same as all the other Content Security Policies, report-to directive is 
+available for private/local IP addresses when the CSP is not valid, for example, unrecognized keyword, invalid URL, etc.
+
+Whether a certain URL is pointing to local/private IP addresses won’t be recorded or reported because of privacy issues.
+
+### [Content-Security-Policy-Report-Only](https://www.w3.org/TR/CSP3/#cspro-header)
+
+Same as above, the report only mode would only report invalid CSPs but not whether the IP address space is correct or not.
+
+## Furthermore
+
+This new CSP can be used not only for navigations but also for various of use 
+cases, for example, HTML image tags, and even to use as an alternative for fetch
+request. In this way, if the target address space is already defined in Content
+Security Policy, then the `targetAddressSpace` fetch option won't be required 
+anymore.
```

Comment:
There are quite a few grammatical errors in this paragraph so I asked AI to refine it.

```suggestion
"This new CSP can be used not only for navigations but also for a variety of use
cases. Examples include HTML image tags and even as an alternative to fetch
requests. If the target address space is already defined in the Content
Security Policy, the targetAddressSpace fetch option will no longer be
necessary."
```